kobo/apps/trash_bin/tests/storage_cleanup/test_attachment_cleanup.py

```diff
@@ -0,0 +1,217 @@
+import pytest
+import uuid
+from unittest.mock import patch
+
+from constance import config
+from constance.test import override_config
+from django.conf import settings
+from django.core.cache import cache
+from django.test import TestCase
+
+if settings.STRIPE_ENABLED:
+    from kobo.apps.stripe.models import ExceededLimitCounter
+
+from kobo.apps.kobo_auth.shortcuts import User
+from kobo.apps.openrosa.apps.logger.models import Attachment
+from kobo.apps.organizations.constants import UsageType
+from kobo.apps.trash_bin.tasks.attachment import auto_delete_excess_attachments, \
+    schedule_auto_attachment_cleanup_for_users
+from kpi.tests.mixins.create_asset_and_submission_mixin import AssetSubmissionTestMixin
+
+
+class AttachmentCleanupTestCase(TestCase, AssetSubmissionTestMixin):
+    def setUp(self):
+        self.owner = User.objects.create(username='owner')
+        self.asset, self.xform, self.instance, self.owner_profile, self.attachment = (
+            self._create_test_asset_and_submission(user=self.owner)
+        )
+
+    def test_auto_delete_excess_attachments_user_within_limit(self):
+        """
+        Test that no attachments are deleted if user is under quota
+        """
+        mock_balances = {
+            UsageType.STORAGE_BYTES: {
+                'effective_limit': 100000,
+                'balance_value': 50000,
+                'balance_percent': 50,
+                'exceeded': False
+            }
+        }
+        with patch(
+            'kobo.apps.subsequences.api_view.ServiceUsageCalculator.get_usage_balances',
+            return_value=mock_balances,
+        ):
+            auto_delete_excess_attachments(self.owner.pk)
+            self.attachment.refresh_from_db()
+            self.assertTrue(Attachment.objects.filter(pk=self.attachment.pk).exists())
+
+    def test_auto_delete_excess_attachments_user_exceeds_limit(self):
+        """
+        Test that attachments are soft deleted when a user is over quota
+        """
+        mock_balances = {
+            UsageType.STORAGE_BYTES: {
+                'effective_limit': 8016257,
+                'balance_value': -6319175,
+                'balance_percent': 178,
+                'exceeded': True
+            },
+        }
+        with patch(
+            'kobo.apps.subsequences.api_view.ServiceUsageCalculator.get_usage_balances',  # noqa
+            return_value=mock_balances,
+        ):
+            auto_delete_excess_attachments(self.owner.pk)
+            self.attachment.refresh_from_db()
+            self.assertFalse(Attachment.objects.filter(pk=self.attachment.pk).exists())
+
+    def test_auto_delete_stops_when_user_is_within_storage_limit(self):
```

Comment:
I think this test name is confusing. I think this test is kind cover by @jamesrkiger's work. I would rather see that the user has _n_ attachments and they are above the limit but after deleting _x_ attachments, they go below their limit and the loop inside `auto_delete_excess_attachments` stop. Then, assert that only the new attachments remain.